cirq-core/cirq/ops/control_values.py

```diff
@@ -12,15 +12,16 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 import abc
-from typing import Union, Tuple, List, TYPE_CHECKING, Any, Dict, Generator, Iterator, Optional
-from dataclasses import dataclass
-
+from typing import Tuple, TYPE_CHECKING, Any, Dict, Iterator, Optional, Sequence, Union
 import itertools
 
+from cirq import protocols, value
+
 if TYPE_CHECKING:
     import cirq
 
 
+@value.value_equality
```

Comment:
does this still gaurntee immutability?

Reply:
Exposing properties instead of public member variables is the Cirq's preferred way of "ensuring immutability" in python. If users are mutating private variables directly, it's on them.